### ISSUE TITLE
Add more hooks in IDCATRDFHarvester

### ISSUE THREAD

**Issue by etj** [ckan/ckanext-dcat#77]
In a use case of ours, we're parsing some metadata in ``parse_dataset()``, and we need to use them *after* the dataset has been stored into the DB.

I added some methods to the ``IDCATRDFHarvester`` interface, that is:
- ``before_update()``
- ``after_update()``
- ``before_create()``
- ``after_create()``

The rationale behind this is:
- ``parse_dataset()``: parse the metadata and add them into the dataset_dict. These metadata fields are not in the dataset schema, so they can't be stored.
- ``before_xxx()``: remove the added metadata from the dict, and save them in some temporary storage
- the ``import`` logic will create/opdate the package
- ``after_xxx()``: take back the metadata fields and do whatever you need to do with them.

Here the definitions of ``before_create`` and ``after_create``; the ``_update`` methods are more or less the same.

```
    def before_create(self, harvest_object, dataset_dict, temp_dict):
        '''
        Called just before the ``package_create`` action.
        It may be used to preprocess the dataset dict.

        Implementations may store some temp values in temp_dict, that will be
        then passed back in the ``after_create`` call.

        :param harvest_object: A ``HarvestObject`` domain object.
        :type harvest_job: object
        :param dataset_dict: The dataset dict already parsed by the RDF parser
                             (and related plugins).
        :type dataset_dict: dict
        :param temp_dict: A dictionary, shared among all plugins, for storing
                          temp data. Such dict will be passed back in the
                          ``after_create`` call.
        :type temp_dict: dict
        '''
        pass

    def after_create(self, harvest_object, dataset_dict, temp_dict):
        '''
        Called just after a successful ``package_create`` action has been
        performed.

        :param harvest_object: A ``HarvestObject`` domain object.
        :type harvest_job: object
        :param dataset_dict: The dataset dict that has just been stored into
                             the DB.
        :type dataset_dict: dict
        :param temp_dict: A dictionary, shared among all plugins, for storing
                          temp data.
        :type temp_dict: dict

        :returns: A string containing an error message, or None. If the error
                  string is not None, it will be saved as an import error,
                  and dataset importing will be rolled back,
        :rtype: string
        '''
        return None
```

**Comment by tdipisa** [comment]
Hi, 
I made a new pull request #81 to introduce a small improvement.

**Comment by metaodi** [comment]
This issue was solved with #78
